pynicotine/plugins/core_commands/__init__.py

```diff
@@ -26,39 +26,637 @@ def __init__(self, *args, **kwargs):
+                "usage": ["<user>"],
+                "usage_private_chat": ["[user]"]
+            },
+            "whois": {
+                "aliases": ["info", "w"],
+                "callback": self.whois_user_command,
+                "description": _("Show info about user"),
+                "disable": ["cli"],
+                "group": _("Users"),
+                "usage": ["<user>"],
                 "usage_private_chat": ["[user]"]
             },
-            "sample": {
-                "description": "Sample command description",
-                "aliases": ["demo"],
-                "disable": ["private_chat"],
-                "callback": self.sample_command,
-                "callback_private_chat": self.sample_command,
-                "usage": ["<choice1|choice2>", "<something..>"],
-                "usage_chatroom": ["<choice55|choice2>"]
+            "browse": {
+                "aliases": ["b"],
+                "callback": self.browse_user_command,
+                "description": _("Browse files of user"),
+                "disable": ["cli"],
+                "group": _("Users"),
+                "usage": ["<user>"],
+                "usage_private_chat": ["[user]"]
+            },
+            "search": {
+                "aliases": ["s"],
+                "callback": self.search_command,
+                "description": _("Start global file search"),
+                "disable": ["cli"],
+                "group": _("Search Files"),
+                "usage": ["<query>"]
+            },
+            "rsearch": {
+                "aliases": ["rs"],
+                "callback": self.search_rooms_command,
+                "description": _("Search files in joined rooms"),
+                "disable": ["cli"],
+                "group": _("Search Files"),
+                "usage": ["<query>"]
+            },
+            "bsearch": {
+                "aliases": ["bs"],
+                "callback": self.search_buddies_command,
+                "description": _("Search files of all buddies"),
+                "disable": ["cli"],
+                "group": _("Search Files"),
+                "usage": ["<query>"]
+            },
+            "usearch": {
+                "aliases": ["us"],
+                "callback": self.search_user_command,
+                "description": _("Search a user's shared files"),
+                "disable": ["cli"],
+                "group": _("Search Files"),
+                "usage": ["<\"user name\">", "<query>"]
+            },
+            "listshares": {
+                "callback": self.list_shares_command,
+                "description": _("List shares"),
+                "group": _("Shares")
+            },
+            "addshare": {
+                "callback": self.add_share_command,
+                "description": _("Add share"),
+                "group": _("Shares"),
+                "usage": ["<public|private|buddy>", "<\"virtual name\">", "<\"folder path\">", ""]  # "" max 3 args
```

Comment:
What is "private"?

Reply:
that's what you originally put in your first cli-commands-test draft.

Reply:
I probably intended to put buddy.

Reply:
Do you know how the core functions are going to be implemented? At the moment, adding and removing shares is not callable as they are handled by either preferences.py or setup assistant.